src/osd/OSD.h

```diff
@@ -515,11 +515,17 @@ class OSDService {
     promote_counter.finish(bytes);
   }
   void promote_throttle_recalibrate();
+  unsigned get_num_shards() const {
+    return m_objecter_finishers;
+  }
+  Finisher* get_objecter_finisher(int shard) {
+    return objecter_finishers[shard].get();
+  }
 
   // -- Objecter, for tiering reads/writes from/to other OSDs --
-  Objecter *objecter;
+  std::unique_ptr<Objecter> objecter;
```

Comment:
can objecter just be a value member to avoid heap allocation entirely?

Reply:
yeah, actually, i thought about this. but i am wondering if we intended to avoid `#include "osdc/Objecter.h"` in `OSD.h`.